File: cxx/src/Creational/Prototype/TestPrototype.cpp
0


//TestPrototype.java - testing the Prototype

class TestPrototype { 
   public static void main(String[] args) {
       System.out.println(
         "Creating a Prototype Factory with " + 
         " a SoupSpoon and a SaladFork");
       PrototypeFactory prototypeFactory = 
         new PrototypeFactory(new SoupSpoon(), new SaladFork());
       AbstractSpoon spoon = 
         prototypeFactory.makeSpoon();
       AbstractFork fork = 
         prototypeFactory.makeFork();
       System.out.println("Getting the Spoon and Fork name:");
       System.out.println("Spoon: " + spoon.getSpoonName() + 
                          ", Fork: " + fork.getForkName());
       System.out.println(" ");       
       System.out.println("Creating a Prototype Factory " + 
                          "with a SaladSpoon and a SaladFork");
       prototypeFactory = 
         new PrototypeFactory(new SaladSpoon(), new SaladFork());
       spoon = prototypeFactory.makeSpoon();
       fork = prototypeFactory.makeFork();
       System.out.println("Getting the Spoon and Fork name:");
       System.out.println("Spoon: " + spoon.getSpoonName() + 
                          ", Fork: " + fork.getForkName());
    }
}      

File: java/t/test/Creational/TestPrototype.java
//TestPrototype.java - testing the Prototype

package test.Creational;


import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Creational.Prototype.*;


class TestPrototype 
{ 

	@Test
   public void prototype() 
	{
       PrototypeFactory prototypeFactory = new PrototypeFactory(new SoupSpoon(), new SaladFork());
       AbstractSpoon spoon = prototypeFactory.makeSpoon();
       AbstractFork fork = prototypeFactory.makeFork();
       org.junit.Assert.assertEquals("Getting the Spoon and Fork name:", 
			  "Spoon: SoupSpoon, Fork: SaladFork", 
			 ("Spoon: " + spoon.getSpoonName() + ", Fork: " + fork.getForkName())
		);


       prototypeFactory = new PrototypeFactory(new SaladSpoon(), new SaladFork());
       spoon = prototypeFactory.makeSpoon();
       fork = prototypeFactory.makeFork();

       org.junit.Assert.assertEquals("Creating a Prototype Factory with a SaladSpoon and a SaladFork", 
       	 "Spoon: SaladSpoon, Fork: SaladFork",
       	("Spoon: " + spoon.getSpoonName() + ", Fork: " + fork.getForkName())
		);

    }
}      
